src/Components/App/App.js

```diff
@@ -34,17 +60,35 @@ class App extends Component {
   }
 
   render() {
+
     return (
       <div className="App">
         <Route 
           exact path='/'>
-          <Homepage />
+          <Homepage 
+            getActivityData={this.getActivityData}
+            allMovies={this.state.movies}
+          />
+        </Route>
+
+        <Route
+          exact path='/:activity'
+          render={({ match }) => {
+            return <BrowsePage 
+            name={match.params.activity}
+            data={ this.state[match.params.activity]}
```

Comment:
I had a hard time rendering the component because I forgot the `return` statement, so don't forget  to add :P